misc/inmanta-workon-register.sh

```diff
@@ -152,47 +186,61 @@ function __get_pip_config_setting {
     return 1
 }
 
+
+function __get_pip_config_setting_all_at_once {
+    # make sure extra index url are formatted correctly (space-separated) e.g. "idx0 idx1 idx2"
+    result=$(
+        "$INMANTA_WORKON_PYTHON" -c "from inmanta.module import Project; project=Project('.', autostd=False); pip_cfg=project.metadata.pip.model_dump(); [(print(k), print(pip_cfg[k]) if pip_cfg[k] is not None else print('')) for k in ['pre','index_url','use_system_config']]; print('extra_index_url'), print(' '.join(pip_cfg['extra_index_url']));" 2> /dev/null
+    )
+    echo "$result"
+    return 0
+}
 function __set_pip_config {
     declare pre
     declare index_url
     declare extra_index_url
     declare use_system_config
 
-    pre=$(__get_pip_config_setting "pre")
-    index_url=$(__get_pip_config_setting 'index_url')
-    extra_index_url=$(__get_pip_config_setting 'extra_index_url')
-    use_system_config=$(__get_pip_config_setting "use_system_config")
+    all_in_one="$(__get_pip_config_setting_all_at_once)" || return
```

Comment:
I don't love this `mapfile` approach but I'm afraid I can't come up with a a feasible alternative. The more this grows the more I wish we generated this whole thing in Python, as Wouter remarked recently, but too late for that.